enforcer/resources/recommendation_store.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import threading
+from typing import Dict, Optional, Tuple
+
+from enforcer.dal.supabase_dal import SupabaseDal
+from enforcer.env_vars import SCAN_RELOAD_INTERVAL
+from enforcer.model import WorkloadRecommendation
+
+
+class RecommendationStore:
+
+    def __init__(self, dal: SupabaseDal):
+        self.dal = dal
+        self.recommendations: Dict[str, WorkloadRecommendation] = {}
+        self.scan_id: Optional[str] = None
+        self._reload_recommendations()
+
+        self.reload_interval = SCAN_RELOAD_INTERVAL
+        self._stop_event = threading.Event()
+        self._reload_thread = threading.Thread(target=self._periodic_reload, daemon=True)
+        self._reload_thread.start()
+
+
+    def _load_recommendations(self, current_stored_scan: Optional[str]) -> Tuple[Optional[str], Optional[Dict[str, WorkloadRecommendation]]]:
```

Comment:
As I understand it, the whole point is that we're mapping a flat list of container recommendations to a hierarchical structure of WorkloadRecommendations containing several container recommendations. If this is correct, I would explicitly document it somewhere as it would have made it easier to understand this code.

Reply:
fixed